Interface with new webAPI

MaxJohansen opened the pull request:
## Changes :balloon:

Signatures are now fetched from the web API instead of from local storage! I've deleted a LOT of leftover resources and code from the old implementation and fixed the failing tests.

## Types of changes

What types of changes does your code introduce? _Mark boxes that apply with an `x`._

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist :heavy_check_mark:

_Mark boxes that apply with an `x`. These can also be filled out after the pull request is created._

- [x] I have read the coding convention [guidelines](https://github.com/Dualog-students/dualog-students.github.io#company-guidelines) and my code follows these.
- [x] Unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

jonjohansen committed:
Saving images!

Interface for saving images has been added
 - Implemented on iOS
 - Permissions have been set for iOS

Views:
Result has gotten two buttons
Save button now has an action, which should run a code-behind function for saving
 NOTE: the name of the image saved in the codebehind should be changed

jonjohansen committed:
Image save!

It now works on android!
Filename sent in is "Navn" for now. This needs to change

jonjohansen committed:
Ground breaking new feature

Images now have names (Only available on an ANDROID near you)!
Also
	Stuff looks clean now

MaxJohansen committed:
Rewrite to use new webAPI

MaxJohansen committed:
Clean up unused assets and fix tests

jonjohansen commented on Diplomatic/ViewModels/TemplatePickerViewModel.cs:
```diff
@@ -1,15 +1,17 @@
 using System.Linq;
 using Diplomatic.Utils;
 
 
```
unnecessary newline 👎

jonjohansen commented on Diplomatic/Views/Templates.xaml.cs:
```diff
@@ -1,11 +1,12 @@
+
     public partial class Templates : ContentPage
     {
-
```
Newline before new block?

MaxJohansen replied:
No

jonjohansen commented on Diplomatic/Views/Templates.xaml.cs:
```diff
@@ -18,8 +19,10 @@ public Templates()
             }
-            var next = new TextFields();
-            next.BindingContext = new TextFieldViewModel((Template)e.SelectedItem);
+            var next = new TextFields
```
newliner 🥈

jonjohansen commented on Diplomatic/Views/TextFields.xaml.cs:
```diff
@@ -12,11 +15,35 @@ public TextFields()
+                string query = string.Join("&", queryParams.ToArray());
+                string final = "https://qri7p78aml.execute-api.eu-west-2.amazonaws.com/dev/" + template.Name.ToLower() + "?" + query;
+                var endpoint = new Uri(Uri.EscapeUriString(final));
+                var next = new Result
```
Newliner 🥉

jonjohansen commented on Diplomatic/Views/TextFields.xaml.cs:
```diff
@@ -12,11 +15,35 @@ public TextFields()
+                {
+                    BindingContext = new ResultViewModel(endpoint, Filename)
+                };
+                await Navigation.PushAsync(next);
```
🥉 Newliner

jonjohansen reviewed (changes requested):
Good but not 💯 🥇

IsakSundeSingh commented on Diplomatic/Models/Field.cs:
```diff
@@ -10,6 +10,8 @@ public class Field
         [JsonIgnore]
         public string Value { get; set; }
+        [JsonIgnore]
+        virtual public bool IsValid => !string.IsNullOrEmpty(Value);
```
This shouldn't be `virtual`. 🥈

IsakSundeSingh commented on Diplomatic/Models/Signature.cs:
```diff
@@ -0,0 +1,20 @@
+        [JsonProperty("id")]
+        public string Id { get; set; }
+        public Uri ImageUri {
+            get {
```
Newlines bro 🥉

IsakSundeSingh commented on Diplomatic/Utils/WebTemplateProvider.cs:
```diff
@@ -22,7 +23,12 @@ public WebTemplateProvider()
                 string content = reader.ReadToEnd();
-                templates = JsonConvert.DeserializeObject<List<Template>>(content);
+                Dictionary<string, Template> rawTemplates = JsonConvert.DeserializeObject<Dictionary<string, Template>>(content);
+                foreach (KeyValuePair<string, Template> entry in rawTemplates)
```
Newline bro :4th_place_medal:

IsakSundeSingh commented on Diplomatic/ViewModels/TemplatePickerViewModel.cs:
```diff
@@ -1,15 +1,16 @@
     {
-        public Template[] TemplateList { get; set; }
+        public Template[] Templates { get; set; }
         public TemplatePickerViewModel()
```
newline

IsakSundeSingh commented on Diplomatic/ViewModels/TextFieldViewModel.cs:
```diff
@@ -1,5 +1,6 @@
 namespace Diplomatic.ViewModels
 {
+    using Models;
     public class TextFieldViewModel
```
newline

IsakSundeSingh commented on Diplomatic/Views/MainPage.xaml.cs:
```diff
@@ -1,35 +1,33 @@
 namespace Diplomatic.Views
 {
+    using ViewModels;
     public partial class MainPage : ContentPage
```
Newline :11st_place_medal:

IsakSundeSingh commented on Diplomatic/Views/MainPage.xaml.cs:
```diff
@@ -1,35 +1,33 @@
         }
 
-        void OnShow ( object sender, EventArgs e)
+        private void OnShow ( object sender, EventArgs e)
```
No space before first function parameter.

IsakSundeSingh commented on Diplomatic/Views/Signatures.xaml.cs:
```diff
@@ -1,27 +1,32 @@
+            template.Signature = (Signature)e.SelectedItem;
             var queryParams = new List<string> { };
-            foreach (var field in template.Fields)
+            foreach (Field field in template.Fields)
```
Newline before this line

IsakSundeSingh commented on Diplomatic/Views/Signatures.xaml.cs:
```diff
@@ -1,27 +1,32 @@
             }
-            var query = string.Join("&", queryParams.ToArray());
-            string final = "https://qri7p78aml.execute-api.eu-west-2.amazonaws.com/dev/" + template.TemplateName.ToLower() + "?" + query;
+            queryParams.Add($"signature={template.Signature.Id}");
```
And after the block <3

IsakSundeSingh commented on Diplomatic/Views/Signatures.xaml.cs:
```diff
@@ -1,27 +1,32 @@
+            string query = string.Join("&", queryParams.ToArray());
+            string final = "https://qri7p78aml.execute-api.eu-west-2.amazonaws.com/dev/" + template.Name.ToLower() + "?" + query;
             var endpoint = new Uri(Uri.EscapeUriString(final));
             var next = new Result
```
This starts a block, so a newline before it.

MaxJohansen replied:
Initializers aren't real blocks to me 👀

IsakSundeSingh commented on Diplomatic/Views/TextFields.xaml.cs:
```diff
@@ -12,11 +16,33 @@ public TextFields()
+                {
+                    BindingContext = new SignaturePickerViewModel(selectedTemplate)
+                };
+                await Navigation.PushAsync(next);
```
Newline.

IsakSundeSingh commented on Diplomatic/Views/TextFields.xaml.cs:
```diff
@@ -12,11 +16,33 @@ public TextFields()
-            await Navigation.PushAsync(next);
+                Template template = ((TextFieldViewModel)BindingContext).SelectedTemplate;
+                var queryParams = new List<string> { };
+                foreach (Field field in template.Fields)
```
Newliner before and after ❤️

IsakSundeSingh commented on Diplomatic/Views/TextFields.xaml.cs:
```diff
@@ -12,11 +16,33 @@ public TextFields()
+                {
+                    BindingContext = new ResultViewModel(endpoint)
+                };
+                await Navigation.PushAsync(next);
```
Come on man... :22nd_place_medal:

IsakSundeSingh reviewed (changes requested):
Fix it

jonjohansen committed:
Fix signatures in view

Resolved scaling issue (Might have to refactor this into a decent scale later)
Aligned signatures to the left.
